New Website, review from [1/3] to [2/3]: Deleting Content

ThomasDelteil opened the pull request:
This is for reviewing in isolation [2/3] while [1/3] is being reviewed and merged

perdasilva committed:
Updates git_init Jenkins utility function to support checking out a particular commit id

perdasilva committed:
Adds artifact repository scripts

perdasilva committed:
Adds CD pipeline framework

perdasilva committed:
Adds static libmxnet release pipeline

perdasilva committed:
Updates CD pipeline

perdasilva committed:
Adds documentation

perdasilva committed:
Updates kvstore functions to use pushd and popd

perdasilva committed:
Throws exceptions instead o magic numbers

perdasilva committed:
Updates artifact repository cli to use --libtype instead of --static or --dynamic

perdasilva committed:
Clarifies ci_utils and cd_utils origin remark

perdasilva committed:
Adds clarifying note on why ubuntu 14.04 is being used for compilation

perdasilva committed:
Removes MXNET_SHA

perdasilva committed:
Removes set_release_job_name

perdasilva committed:
Adds license headers

perdasilva committed:
Updates artifact repository to expect licenses

perdasilva committed:
Moves ci/cd to cd directory

perdasilva committed:
Takes downstream job name from environment

perdasilva committed:
Updates order of parameters

perdasilva committed:
Updates job type parameter to dropdown

perdasilva committed:
Adds libmxnet feature extraction code comments

perdasilva committed:
Removes ccache setup from static build

ptrendx committed:
Allow operators with multiple outputs in get_atomic_symbol (#15740)

* Allow operators with multiple outputs in get_atomic_symbol

* Added unittest

ThomasDelteil committed:
removing old content

Co-authored-by: Thomas Delteil <thomas.delteil1@gmail.com>
Co-authored-by: Aaron Markham <markhama@amazon.com>

haojin2 committed:
numpy-compatible split upstream (#15841)

ChaiBapchya committed:
Large tensor support for random ops (#15783)

* random ops

* replace array with uniform

* remove dtype

* randn add

* add multinomial

* multi,randn small fix

* add negative bino

* fix memory issue - Failed to allocate CPU Memory

* Trigger notification

* linting fix

* Trigger notification

ChaiBapchya committed:
Add Large Tensor Support for Sequence, NN Ops  (#15807)

* sequence_last, sequence_reverse, sequence_mask

* working softmax_cross_entropy

* fix linting, add index_copy

* add softmax output

* add leaky relu

* add pooling

* add layernorm

* add dropout, activation, batchnorm and update layernorm

* address comments to remove some comments

* handling imports

haojin2 committed:
Numpy-compatible concatenate upstream (#15894)

* numpy-compatible concatenate upstream

* extend ci deadline

haojin2 committed:
Numpy-compatible concatenate upstream (#15894)

* numpy-compatible concatenate upstream

* extend ci deadline

haojin2 committed:
Numpy-compatible concatenate upstream (#15894)

* numpy-compatible concatenate upstream

* extend ci deadline

haojin2 committed:
Numpy-compatible concatenate upstream (#15894)

* numpy-compatible concatenate upstream

* extend ci deadline

anirudh2290 committed:
Fix ConcatType backward type inference (#15829)

* Fix ConcatType and add test

* Remove return false

* Change error message

* Run RNN test only when CUDNN enabled

* set default context for test_contrib_amp

ChaiBapchya committed:
[OpPerf] Profiler flag for Python, Cpp  (#15881)

* first

* flag for timing python

* delete old file

* added profiler example

* add operator name to python profile output

* add other pending files with profiler flag

* change cpp to native - for user-facing api

* fixes

* Trigger notification

* Trigger notification

* Trigger notification

* markdown update for python

samskalicky committed:
added check for empty params file and unknown param (not arg/aux) (#15917)

* added check for empty params file and unknown param (not arg/aux)

* changed exception to warning for unknown params

* removed unnecessary MXNetError import

* added warning message is params is empty

* fixed print

* fixed formatting

* missing paren

zachgk committed:
Remove Scala package test on build (#15915)

Various build improvements to mirror pip build

ChaiBapchya committed:
Add power, exponent, log ops large tensor support (#15794)

* power, exponent, log ops

* lint

* Trigger notification

* Trigger notification

* Trigger notification

ChaiBapchya committed:
[Opperf] Filter out deprecated ops (#15541)

* normal, uniform ops

* Revert "normal, uniform ops"

This reverts commit f8d6f9583da859e1f5053c2689eafae5c3031553.

* filter out deprecated ops

* Trigger notification

* additional deprecated ops

* Trigger notification

* Trigger notification

* Trigger notification

ZhennanQin committed:
Add disable attr to subgraph property (#15926)

DickJC123 committed:
Refactor for windows CI 'out of heap space' errors (#15922)

* Speed up test_random.py:test_shuffle to get past CI timeouts.

* Fix flakey test_operator.py:test_laop_6.

* Break up broadcast_reduce_op_value.{cc,cu}

* Revert "Fix flakey test_operator.py:test_laop_6."

This reverts commit 479ba3860884402914ef7df2fc5a031fed5b0f37.

* Break up elemwise_unary_op_basic.{cc,cu}

haojin2 committed:
Numpy-compatible stack upstream (#15842)

larroy committed:
Re-enable flaky test_prelu (#15777)

Fixes #12885

hzfan committed:
declare explicitly the tblob default assign operator and copy constructor (#15937)

ChaiBapchya committed:
[OpPerf] Handle positional arguments (#15761)

* handle args

* intermediate, error with getting 2 values for data param for other ops

* handle args

* None type issue

* try

* indent fix

* lint fix

* Trigger notification

* Trigger notification bcoz validation/edge passed but shows pending

ChaiBapchya committed:
[OpPerf] Take care of 4d param  (#15736)

* take care of 4d

* Trigger notification

* Trigger notification

* Trigger notification

* Trigger notification

* Trigger notification

* Trigger notification

* Trigger notification

* Trigger notification

* Trigger notification bcoz validation/edge passed but shows pending

access2rohit committed:
removing unnecessary int64 C apis that were added to support Large Tensors and Vectors (#15944)

xziya committed:
Discard needless test cases in `test_convolution_independent_gradients` (#15939)

* Drop needless num_bases in test of conv

* Trigger CI

* Re-trigger CI

access2rohit committed:
creating ndarray directly using mxnet ndarray primitives to reduce memory footprint of tests for topk, sort and argsort (#15900)

ChaiBapchya committed:
Benchmark doc fix (#15769)

* Update pre-req for opperf

* Update README.md

* correct command to import binary broadcast

* no such op called nd.sub, it is nd.subtract

* Trigger notification

* Trigger notification

access2rohit committed:
Adding tests to verify support for Large Tensors in additional Ops along with new C_Apis supporting 64bit indexing (#15895)

arcadiaphy committed:
fix naive engine for multi-threaded inference (#15574)

* fix naive engine

* remove thread local

* trigger CI

xinyu-intel committed:
[Quantization]support exclude operators while quantization (#15910)

* skip unregistered op and add exclude_op_names

* wrap a function

* add testcase for excluded op

* skip test for native cpu since all op will be excluded

* trigger

* trigger again

* revert check inner node op name

* test subgraph

* trigger

* trigger

xidulu committed:
[Numpy] Numpy behavior random.uniform() (#15858)

* numpy behavior uniform sampling implemented

* improve code style according to review comment

* remove rebundant namespace

ptrendx committed:
Softmax optimization for GPU (#15545)

* Softmax optimization

* Fix lint

* Unifying softmax with length with regular softmax

* Fix lint

* Fixes from review

* Making less templated kernels

* Unifying softmaxgrad and softmaxwithlengthgrad

* Better gradient of softmax

* Dividing softmax.cc into multiple files

* Fix

* Trigger CI

* Moving get_rows_per_block to common place

* Fix

* Fix lint

* Actually fix lint

stu1130 committed:
[Dependency Update] [Doc] move the general prerequisite software to the top (#15896)

* move the general prerequisite software to the top

* retrigger CI

ciyongch committed:
[MKLDNN]Support fullyconnected and element-wise ops fusion (#15950)

* enable fc fusion with more eltwise ops

* refactor codes

* Add test cases for fc fusion pattern

* cleanup codes

* remove a_max limitation for bounded_relu

hzfan committed:
Tvm broadcast backward (#15938)

* tvm broadcast backward

* dispatch by req

* pad for broadcast to a larger dim

stu1130 committed:
Disable test coverage for Clang MKLDNN (#15977)

* disable test coverage

* disable the job

ptrendx committed:
Fix get_rows_per_block (#15979)

apeforest committed:
Fix a memory misalignment in topk operator (#15948)

* fix alignment

* use correct type for shape index

* clean up unnecessary space in topk

* fix lint

* add additional temp space

* address reviewer comment

* fix incorrect nidex type

ElaineBao committed:
update support MKLDNN BN conditions (#15870)

* update support MKLDNN BN conditions

* add bn test case for channel size = 8

* fix bn gradient with use_global_stats

* rm useless comments

* fix mkldnn bn output when use_global_stats

* fix lint

* retrigger ci

* retrigger ci again

* retrigger ci again 2

* retrigger ci again 3

* retrigger ci again 4

leezu committed:
Remove requests_failed_to_import handling

We already required requests package. If a user is missing it nevertheless, fail
with a standard error message instead of some hard to understand error that
requests_failed_to_import object does not have a get method..

Example https://github.com/dmlc/gluon-nlp/issues/892

haojin2 committed:
np elemwise unary ops upstream (#15831)

roywei committed:
remove Julia cat image for license issue (#15964)

* remove cat image

* trigger ci

anirudhacharya committed:
Add fp16 support for topk (#15560)

* fp16 for topk

* indentation and mem alloc

* fix cuda code

* fix cuda sort routine for half precision

* remove ambiguous half type

zachgk committed:
Fix Nightly Maven GPU (#15989)

ChaiBapchya committed:
Add symbol api for randn and fix shape issue for randn ndarray and symbol api (#15772)

* symbol api for randn, fix shape issue for randn ndarray and symbol api

* address comments; add unit test for symbol and check for NDArray

* shape issue

* address comments - unit test, doc fix, etc

* remove prints, test randn for symbol and multidistribution

* Trigger notification bcoz data.mxnet.io failed

* removed stray file

* Trigger notification bcoz R/cpu failed

* add to symbol doc

ChaiBapchya committed:
Decouple dtype from shape for Random multinomial (#15980)

* change int to index t and remove the check

* fix lint

* Trigger notification

* Trigger notification coz of laop crash

TaoLv committed:
Fix dtype inference in arange_like operator (#15930)

* fix dtype in arange_like operator

* add unit test

ciyongch committed:
[MKLDNN] Fix out of bound access of req vector (#16000)

* remove unnecessary creation of req vector

* retrigger CI

marcoabreu committed:
Disable laop_6 (#15976)

marcoabreu committed:
Update CODEOWNERS. (#15972)

* Update CODEOWNERS

* retrigger ci

* Update CODEOWNERS

ElaineBao committed:
add uint8 bn mkldnn implementation (#16003)

* add uint8 bn mkldnn implementation

* update test case for uint8 bn

* fix lint

* update test with gpu

* add comment for quantization

* fix quantized_bn test

* fix quantize_model_with_forward test

vishaalkapoor commented:
The readme is the only non-deleted file with deltas and it looks good. LGTM!

vandanavk committed:
Correct ONNX documentation (#15914)

anirudh2290 committed:
Add AMP Conversion support for BucketingModule (#15528)

* Add AMP Conversion support for bucketing module

* Add bucketing module changes

* Add load_aprams

* Add bucketing module conversion

* Fix unused arg

ChaiBapchya committed:
Add Median,p50,p99 to python profiler (#15953)

* median,avg,p90,p99

unused var remove

add parameter

missing arg

add mean, median, p90,p99 to markdown

markdown

* Trigger notification coz of gpu windows fail

* Trigger notification

* Trigger notification

* add _time

xidulu committed:
[Numpy] random.randint() implemented (#15956)

* randint completed

* doc fixed

* tiny bugs fixed

* Update runtime_functions.sh

* change CI config back

access2rohit committed:
Added tests to verify Large Vector Support for initial set of ops  (#15943)

* Adding tests to verify support for Large Tensors in additional Ops along with new C_Apis supporting 64bit indexing

* removing skipped tests

* enabling Large Index support for slice and softmax

* removing tests not required for vector testing

zoeygxy committed:
Refines NDArray indexing and adds numpy ndarray indexing [READY FOR REVIEW] (#15942)

* Completed NDArray indexing fix and numpy ndarray indexing.

* Style and test fixed

* Retrigger CI

larroy committed:
Windows cmake flags cleanup (#16013)

larroy committed:
Improve diagnose.py to display environment variables (#15715)

anirudh2290 committed:
Disable flaky test in test_amp_conversion (#16031)

ZhennanQin committed:
Improve quantization flow (#15961)

* Add mkldnn imlementation for quantized flatten and smart quantize mode

Change-Id: Id9d7504890852faf4e84fdcd66585d1fd78beeb2

* Add calibrate op

Change-Id: I4c82f64dbef501d2560f7ffee93991119a66a5ee

* Fix merge

Change-Id: I068df3d4f3309bc9b950a3b869da9407282c8577

* Fix lint

Change-Id: Ia38369d31c33d0f76a671275910729dfce693950

* Run CI

* Run CI

* Update test_quantization.py

* Run CI

* Fix CI

Change-Id: I7479327db5ebc7c57b7bd810a67d2b765c820534

* Fix CI

Change-Id: I4273938cb972c12b8f43dbd95c736a7d32df040e

* Fix CI

Change-Id: I80b47bd1d95520a7cd78cacbbc1a85fe0900123d

* Fix CI

Change-Id: I56542470010e7bc403f62dc8a8991c2fb58d229e

* Fix CI

Change-Id: If8482fe4da2f3d627dd3cbac8795e021a09a441f

* Fix GPU

Change-Id: Ic239cbf7aa3d111f2895badd1cac196fce6a1b86

reminisce committed:
Port ops from np branch (#16018)

* Add maximum, minimum, swapaxes, argmax, clip in python

* Add backend

* Fix pylint

* Add unit test decorators back

* Fix gpu compile

* Add np.random.normal and npx.seed

* Expose seed through npx.random

* Add rtol atol in seed testing

tlby committed:
[MXNET-1399] multiclass-mcc metric enhancements (#14874)

* multiclass-mcc metric enhancements

 * Rename metric from "PCC" to "mMCC" because though the math is derived
   from Pearson CC, it's utility is as a multiclass extension of Mathews CC.
 * Harden mx.metric.mMCC.update to more variations of input format, similar to
   mx.metric.Accuracy.update.

* Harden mx.metric.PCC.update to more variations of input format, similar to mx.metric.Accuracy.update.

* Enhance testcases for mx.metric.PCC.

vandanavk committed:
[MXNET-895] ONNX import/export: TopK (#13627)

* ONNX import: TopK

* ONNX export: TopK

* Add test for TopK

* Rebasing

* Fix a comment and rebase

* Re-trigger CI

* Re-trigger CI

haojin2 committed:
numpy-compatible cumsum upstream (#15924)

RuilinZhuIntel committed:
Update README.md (#16035)

Added mxnet 1.5.0 release hyper link and description into What's new in readme.md

ElaineBao committed:
Add RROIAlign (#16017)

* add RROIAlign

* add test case for RROIAlign

* move rroi align op to contrib

* skip gpu test for rroi align

* add omp parallel

* fix lint

* update omp with num_threads

* fix extra copy

haojin2 committed:
NumPy-compatible infrastructure on Gluon (#16024)

* numpy infra residual part

* Add test

* Fix sanity

* Fix flaky test

ElaineBao committed:
[MKLDNN] fix uint8 batch norm memory misuse (#16034)

* fix uint8 bn reorder memomry

* update resnet152-v2 int8 acc

stu1130 committed:
Disable test coverage of C++ codebase on CI  (#15981)

* disable test coverage on ci

* address the comment

* retrigger CI

reminisce committed:
[OP] Support range as advanced index for ndarrays (#16047)

access2rohit committed:
Added more tests for Large Indices (#15960)

* Adding tests to verify support for Large Tensors in additional Ops along with new C_Apis supporting 64bit indexing

* removing skipped tests

* removing tests not required for vector testing

* Adding more tests for Large Indices and adding support for Large Indices in one_hot operator

* Re-Trigger build

stu1130 committed:
Numpy compatible max min (#16046)

* numpy compatible max min

* retrigger CI

perdasilva committed:
Fixes openblas installation for static build

eric-haibin-lin committed:
[Dev] update ps-lite dependency (#15936)

* update ps-lite

* Revert "update ps-lite"

This reverts commit dfaece914842a8923892f7ed190ebb271c9d68e8.

* update ps-lite dependency

gigasquid committed:
Fix flaky clojure profile test (#16058)

* Fix flaky clojure profile test

* disable profiler test

gyshi committed:
fix test_pick test time  is too long (#16066)

kshitij12345 committed:
[fix] Support nullop in `transpose` (#15865)

* support nullop transpose

* add relevant test

* retrigger CI

* update error message

* retrigger CI

* retrigger CI

* retrigger CI

* retrigger CI

* retrigger CI

* retrigger CI

haojin2 committed:
NumPy-compatible Mean, Std and Var (#16014)

* NumPy-compatible mean upstream

* NumPy-compatible std and var

* fix sanity and flakiness of tests

Zha0q1 committed:
adding "total" (total time) to profiler aggregate stats sorting criteria (#16055)

* Update aggregate_stats.h

* add 'total' to sorting criteria

xinyu-intel committed:
fix flaky test (#16074)

mseth10 committed:
Graph Partition API (#15886)

* API to trigger partitioning

* pre- and post-partition functions for subgraph property

* adding infer shape type before partition

* modifying pre/post-partition declaration

* adding support for infer shape type before partition

* passing kwargs down to pre/post partition funcitons

* move InferForwardAttrs to common/

* Addressing github comments

* refactoring to enable infer shape/type without storage type

* check if subgraph rejected by subgraph property

* adding description

* setting graph attribute context from args

* adding unit test for optimize_for with default backend

* fixing args access

* removing options_map from PostPartition

* addressing PR comment

* adding logs about status of subgraph node creation

* allowing partial infer shapes

* added context argument back to optimize_for and removed args context validation

* fixed spacing and dev_type

* fixing lint

* reorganized args list to optimize_for

* fixing spacing

* dereferencing dev_type

reminisce committed:
Add fluent methods mean, std, var for ndarray (#16077)

* Add fluent methods mean, std, var for ndarray

* Fix sanity

gyshi committed:
fix some test files test time is too long (#16067)

haojin2 committed:
new raise mode for nd.take and fix backward for wrap mode (#15887)

access2rohit committed:
Typedef cleanup (#15899)

* change mx_uint->int32_t

* changing mx_int64 -> int64_t

* change mx_float -> float

* maintining typedefs for other language bindings

* Re-Trigger build

TaoLv committed:
add KEY for Tao Lv (#16081)

stu1130 committed:
numpy multinomial op (#15878)

* numpy multinomial op

* address the comment

* retrigger CI

* retrigger CI

xziya committed:
MKL-DNN RNN checks NDArray version (#16071)

* MKL-DNN RNN checks NDArray version

* Add UT

* Use default_context()

gyshi committed:
add numpy operator remainder (#16080)

AlexChalk committed:
Update readme and project.clj comment (#16084)

ChaiBapchya committed:
Add Large tensor vector test cases (#15941)

* add random ops

* add shuffle to test large array

* shape evaluation after value check

* add log, exponent, power ops

* fix sequence reverse issue in test_large_array and add sequence ops to test_large_vector

* add binary arithmetic

* fix lint, minor mistakes in large_array; add nn op to tensor

* Trigger notification coz of test_operator.test_laop_6 error

* Trigger notification coz of test_operator.test_laop_6 error

* Trigger notification bcoz R failures

* address comments

* normal distribution assert statement fix; randint dtype check

* correct layernorm and shuffle

* layer norm numpy flaky hence removed, dropout shape fix

* comment not working ops

* fix multi

* Trigger notification

* fix seq reverse, uncomment seq mask as it works

* index fix and uncomment test

* index fix

* seq_reverse index fix

* uncomment seq reverse test and handle static typecasts

* removing commented ops

* resolve merge conflict

* teardown, lint, remove redundant functions

* fix shape assertions and randint low,high

* remove waits, add teardown to large_array, change randint assert in large array

chongruo committed:
typo in docs (#16094)

* typo for NCCL

* retrigger CI

apeforest committed:
remove 'foo' and other print msg from test (#16088)

* remove 'foo' and other print msg from test

* address review comment

* re-trigger CI

yzhliu committed:
Enable tvm_op for ci (#15889)

* enable tvm_op for ci

* specify python3 bin

* move rpath to top

* move tvm op dep forward

* add ldd debug info

* add libtvm_runtime.so to mx_lib_cython

* add ldd debug for py3

* fix libtvm_runtime path for cmake

* cp libtvm_runtime.so when make rpkg

* add libtvm_runtime.so to scala-pkg

* add python3 to cmake in unix-gpu build

* hack: add cuda to ld_path in cmake

* add LD_LIBRARY_PATH into cmake tvm op

* add /usr/local/cuda/compat to Dockerfile.build.ubuntu_gpu_cu101 LD_LIBRARY_PATH

* remove unused codes

* remove USE_TVM_OP from build_ubuntu_cpu_large_tensor

apeforest committed:
Test large vector mean operator and fix a few bugs (#16079)

* add test_mean and fix a few potential bugs

* address reviewer commnet

* fix and refactor test

* fix topk test

* address reviewer comment

kshitij12345 committed:
Fix gradient tensor mutate in `{adam/ftrl/rmprop/rmspropalex}_update`. (#15768)

* update code to fix #15759

* add relevant test

* re-add the removed conditional dispatch

* fix grad mutate for ftrl_update

* add test for ftrl_update

* fix grad mutate for rmspropalex_update

* add test for rmspropalex_update

* use KERNEL_ASSIGN in RMSPropAlexUpdateKernel.

* fix grad mutate for rmsprop_update

* add test for rmsprop_update

* add more optimizers for mutation test

* retrigger CI

* retrigger CI

* retrigger CI

* retrigger CI

* address comments.

* refactor code.

* retrigger CI

* retrigger CI

* retrigger CI

anirudh2290 commented on docs/README.md:
```diff
@@ -20,81 +20,160 @@
+
+MXNet's Python documentation is built with [Sphinx](http://www.sphinx-doc.org) and a variety of plugins including [pandoc](https://pandoc.org/), and [recommonmark](https://github.com/rtfd/recommonmark).
+
+More information on the dependencies can be found in the [CI folder's installation scripts](https://github.com/apache/incubator-mxnet/tree/master/ci/docker/install/ubuntu_docs.sh).
```
nit: when i look up the file https://github.com/apache/incubator-mxnet/blob/master/ci/docker/install/ubuntu_docs.sh , it says ```pip3 install -r /work/docs_requirements```. maybe we should make it clear where this is coming from. either in the script or here.

ThomasDelteil committed:
[WIP] New Website: New Docs [1/3] (#15884)

* Adding new website files

Co-authored-by: Thomas Delteil <thomas.delteil1@gmail.com>
Co-authored-by: Aaron Markham <markhama@amazon.com>

* add license header

* fixing links and url + css

* fixing tutorials

* exclude python_docs folder when building prod site

This allows things to coexist until a following PR deletes this file.

* license header fixes

* exclude static site from license header check

* exclude static site and sphinx theme

* exclude static_site and cpp_docs from sphinx

* add licenses and excludes

doronsinger committed:
Speed up group executor (#16069)

* Speed up group executor

Current implementation is O(n^2), this implementation is O(n)

* Speed up group executor

Current implementation is O(n^2), this implementation is O(n)

* CI

eric-haibin-lin committed:
[DOC] Fix doc for nn.Embedding, nn.Dense and nd.Embedding (#15869)

* Update basic_layers.py

* Update indexing_op.cc

kshitij12345 committed:
[MXNET-978] Higher Order Gradient Support `arctan`, `arctanh`, `radians`. (#15531)

* support arc{tan/tanh} for higher order grad

* add relevant tests

* add new abstraction for Node operations

* support radians for higher order grad

* add test for radians

* changes

* use NodeOp for arctan.
* update few comments.
* update few variable names.

* rename grad_x to x_grad

* update comments

* move node_op_util.h to src/nnvm

* address comments

* rename NodeOp to NodeOpGen.
* rename Op to op.

* fix file description

larroy committed:
Revert accidental change to CMakelists (#16040)

anirudh2290 committed:
Not to search for coverage files when none exist (#16107)

kshitij12345 committed:
[MXNET-978] Higher Order Gradient Support `sqrt`, `cbrt`. (#15474)

* support sqrt, cbrt for higher order grad

* add relevant tests

* remove unnecessary variable

kshitij12345 committed:
[MXNET-978] Higher Order Gradient Support `clip`, `dropout`. (#15746)

* support clip, dropout for higher order grad

* add relevant tests

* retrigger CI

apeforest committed:
Fix unary operator ceil/floor/trunc when data type is integer (#14251)

* fix integer precision loss due to casting to float

* add test

* update comment

eric-haibin-lin committed:
[Dataset] Add take, filter, sample API to dataset (#16078)

* first commit

* fix lint

* add more test  for transformed data

* address CR

* Update dataset.py

* use sampler to implement

* Update dataset.py

iblislin committed:
julia: rename build env var `MXNET_HOME` to `MXNET_ROOT` (#15568)

* julia: rename build env var `MXNET_HOME` to `MXNET_ROOT`

- Add MXNET_LIBRARY_PATH support

Ref: https://github.com/apache/incubator-mxnet/pull/15561

* backward compatibility

terrytangyuan committed:
[DOC] Consistent capitalization: mxnet -> MXNet, scala -> Scala (#16041)

anandj91 committed:
[MXNET-1294] Add KVSTORE PushPull API (#15559)

* Switch to latest ps-lite

* Add PushPull API

* Add PushPull test cases

AlexChalk committed:
Fix failing tests (#16117)

xidulu committed:
[Numpy] Random.choice implemented (#16089)

* imperative choice done

* unit test done

* expose take to np internal

* style fixed

* style fixed

* style problems fixed

* remove out parameter and fix style

* fix syntax error

zachgk committed:
Update python dependencies (#16105)

ThomasDelteil committed:
adding license headers (to be removed)

aaronmarkham committed:
fix headers

aaronmarkham committed:
restore file for legacy builds

access2rohit committed:
Reducing memory footprint of one_hot for Large Array Testing (#16136)

* reducing memory footprint of one_hot for Large Tensor

* removing one_hot from large_vector and changing large_array test to incorporate large vector part of the test

reminisce committed:
Fix sample.normal shape inference

anirudhacharya committed:
Fixes NAG optimizer #15543 (#16053)

* fix update rules

* readable updates in unit test

* mom update

kshitij12345 committed:
[MXNET-978] Higher Order Gradient Support `sinh`, `cosh`. (#15412)

* add higher order support for sinh cosh

* add relevant tests

* update comments

* retrigger CI

aaronmarkham committed:
Revert "julia: rename build env var `MXNET_HOME` to `MXNET_ROOT` (#15568)" (#16147)

This reverts commit 75ee1e9f434383eff9b250f32b10c89e721f66d3.

Vikas-kum committed:
Fixing build for gluon estimator test, including libtvm in pack libs (#16148)

* Fixing build for gluon estimator test, including libtvm in pack libs

* Fixing html build for nightly

gyshi committed:
Numpy add numpy op indices (#15837)

* add numpy op indices

* re-trigger ci

* fix cpplint

iblislin committed:
julia: fix `mx.forward` kwargs checking (#16138)

close https://github.com/dmlc/MXNet.jl/issues/431

eric-haibin-lin committed:
Update env_var.md (#16145)

perdasilva committed:
CD Fixes (#16127)

* Ignore load lib test in CD jobs

* Removes cu80 and adds cu101 support to CD builds

* Disable cython in CD python tests

* Updates CD documentation to reflect variant changes

ThomasDelteil committed:
merge the conflicts

ElaineBao committed:
avoid test relu at the origin due to discontinuous gradient (#16133)

* avoid test relu at the origin due to discontinuous gradient

* retrigger CI

ThomasDelteil committed:
empty commit to trigger CI

ThomasDelteil committed:
update submodule to correct one

aaronmarkham committed:
delete website check

aaronmarkham committed:
add placeholder website check

aaronmarkham committed:
add placeholder website check, for reals this time

reminisce committed:
Fix remaining errors reported by D2L (#16157)

* Fix

* Fix ssd

* Add test for np.empty

* Add np.linalg.norm

* Fix indexing bug

* Improve doc

hzfan committed:
[Numpy] Numpy copysign (#15851)

* add numpy compatible copysign

* fix scalar op registration error

* add test

ZhennanQin committed:
Update dmlc-core (#16149)

Change-Id: I472fb7bbffc16ed8c36494ab49838b08c59b2f12

sxjscience committed:
use 1E-4 in groupnorm test(#16169)

tingying2020 committed:
numpy operator ravel, derive from reshape (#16016)

* it is the same as reshape(x, -1)

* register reshape with prefix _npi_

* fix format error

* edit examples in doc

* fix error in review

* add out in wrapper

* remove out

* test data type and add order

* change order check

* remove redundant test and add docstring about order

perdasilva committed:
Adds dynamic libmxnet to CD pipeline (#16163)

* Adds dynamic libmxnet release pipeline

* Updates default variants on CD pipeline

ChaiBapchya committed:
Sequence last fix (#16156)

* seq last fix

* index tensor to have int64

* fix dtypes

* revert unnecessary changes

* if seq len not passed, pass int64 dtype

* dtype comment

* use int32 or int64 as index dtype based on build flag

* Trigger notification

* Trigger notification

* lint fix

Vikas-kum committed:
fixing test for model compatibility checker (#16159)

* fixing test for model compatibility checker

* fake change

Vikas-kum committed:
Tutorials nighly fix (#16179)

* Fixing tutorials test failures

* profier set state

* modifying Jenkins for fast run

* chanign comments

* more test comment for faster run

* changing comments from

* more comments

* profiler tutorial test fixes

* uncommenting all tutorial tests

* enabling all tests for nightly

* removing parallel keyword from tutorials test

* updating comments

* Added github issue link in the comment

aaronmarkham committed:
fix merge conflict

aaronmarkham committed:
Merge branch 'ThomasDelteil-new_website_delete_content' into new_website_delete_content

ThomasDelteil committed:
trigger CI

ThomasDelteil committed:
trigger PR
